jvm/src/test/scala/org/scalamock/test/specs2/ConcurrencyTest.scala

```diff
@@ -20,26 +20,24 @@
 
 package org.scalamock.test.specs2
 
-import scala.concurrent.Await
+import scala.concurrent.{Await, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
-import scala.concurrent.duration.Duration
-import scala.concurrent.duration.SECONDS
-
-import org.scalamock.specs2.MockContext
+import scala.concurrent.duration.{Duration, SECONDS}
+import org.scalamock.specs2.IsolatedMockFactory
 import org.scalamock.test.mockable.TestTrait
 import org.specs2.mutable.Specification
 
-class ConcurrencyTest extends Specification {
+class ConcurrencyTest extends Specification with IsolatedMockFactory {
```

Comment:
Using this instead of a new MockContext for each test seemed to be fundamental to writing a test that failed reliably